app/components/works/show/related_contents_show_component.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Works
+  module Show
+    # Component for rendering related contents on the work show page
+    class RelatedContentsShowComponent < ApplicationComponent
+      def initialize(work_presenter:)
+        @work_presenter = work_presenter
+        super()
+      end
+
+      attr_reader :work_presenter
+
+      delegate :related_links, to: :work_presenter
+
+      def related_works
+        work_presenter.related_works_attributes.select { |related_work| related_work.to_s.present? }
```

Comment:
If you use `work_presenter.related_works` here, I think you can drop the explicit `to_s` in the block? Worth a quick try.